src/content/reference/react-dom/components/form.md

```diff
@@ -0,0 +1,445 @@
+        }}
+      >
+        <input type="text" name="message" placeholder="Hello!" />
+        <button type="submit">Send</button>
+      </form>
+    </div>
+  );
+}
+
+export default function App() {
+  const [messages, setMessages] = useState([
+    { message: "Hello there!", sending: false, key: 1 }
+  ]);
+  async function sendMessage(formData) {
+    const sentMessage = await deliverMessage(formData.get("message"));
+    setMessages([...messages, { message: sentMessage }]);
+  }
+  return <Thread messages={messages} sendMessage={sendMessage} />;
+}
+```
+
+```js actions.js
+export async function deliverMessage(message) {
+  await new Promise((res) => setTimeout(res, 1000));
+  return message;
+}
+```
+
+
+```json package.json hidden
+{
+  "dependencies": {
+    "react": "experimental",
+    "react-dom": "experimental",
+    "react-scripts": "^5.0.0"
+  },
+  "main": "/index.js",
+  "devDependencies": {}
+}
+```
+
+</Sandpack>
+
+### Handling form submission errors {/*handling-form-submission-errors*/}
+
+In some cases the function called by a `<form>`'s `action` prop throw an error. You can handle these errors by wrapping `<form>` in an Error Boundary. If function called by a `<form>`'s `action` prop throws an error the fallback for the error boundary will be displayed.
+
+<Sandpack>
+
+```js App.js
+import { ErrorBoundary } from "react-error-boundary";
+import { action } from "./actions.js"
+
+export default function Search() {
+  return (
+    <ErrorBoundary fallback={<p>There was an error while submitting the form</p>}>
+      <form action={action}>
+        <input name="query" />
+        <button type="submit">Search</button>
+      </form>
+    </ErrorBoundary>
+
+  );
+}
+```
+
+```js actions.js hidden
+export async function search() {
+  return new Promise((_, reject) => reject())
+}
+```
+
+```json package.json hidden
+{
+  "dependencies": {
+    "react": "experimental",
+    "react-dom": "experimental",
+    "react-scripts": "^5.0.0",
+    "react-error-boundary": "4.0.3"
+  },
+  "main": "/index.js",
+  "devDependencies": {}
+}
+```
+
+</Sandpack>
+
+### Processing forms differently with different submit buttons {/*processing-forms-differently-with-different-submit-buttons*/}
+
+Forms can be designed to handle multiple submission actions based on the button pressed by the user. Each button can be associated with a distinct action or behavior. When a user taps a specific button, the form is submitted, and a corresponding action, defined by that button's attributes and action, is executed. For instance, a form might allow users to either save their input as a draft or submit it for review with separate buttons.
+
+<Sandpack>
+
+```js App.js
+export default function Search() {
+  function publish(formAction) {
+    const content = formAction.get("content");
+    const button = formAction.get("button");
+    alert(`'${content}' was published with the '${button}' button`);
+  }
+
+  function save(formAction) {
```

Comment:
the arguments here are `formData` not `formAction`, right?

Reply:
good catch, fixed.